macOS Sonoma 14.1.2 `pip install pygraphviz` fails (missing lib cdt)

This is the only solution I found at the very bottom of unrelated issue (posting another issue hoping to help someone else):
https://github.com/pygraphviz/pygraphviz/issues/11#issuecomment-1876025261

If this is correct for macOS Sonoma, can the package author please add this to install docs?

This works for me as of 2024-01-11 on M1 MacBook Pro with Sonoma 14.1.2 / Python 3.11.6. (Until Apple breaks something again, LOL.)
```
brew install graphviz

# Open new Terminal window to be sure. Then activate virtual environment:

.venv/bin/activate

pip install --no-cache-dir \
  --config-settings="--global-option=build_ext" \
  --config-settings="--global-option=-I$(brew --prefix graphviz)/include/" \
  --config-settings="--global-option=-L$(brew --prefix graphviz)/lib/" \
  pygraphviz
```

I was getting the error: `ld: library 'cdt' not found` and could not find a way to install that. Here is full stack trace (running in venv):

```
pip install pygraphviz             main 

Collecting pygraphviz
  Using cached pygraphviz-1.12.tar.gz (104 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: pygraphviz
  Building wheel for pygraphviz (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for pygraphviz (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [56 lines of output]
      running bdist_wheel
      running build
      running build_py
      creating build
      creating build/lib.macosx-14-arm64-cpython-311
      creating build/lib.macosx-14-arm64-cpython-311/pygraphviz
      copying pygraphviz/scraper.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz
      copying pygraphviz/graphviz.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz
      copying pygraphviz/__init__.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz
      copying pygraphviz/agraph.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz
      copying pygraphviz/testing.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz
      creating build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/test_unicode.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/test_scraper.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/test_readwrite.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/test_string.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/__init__.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/test_html.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/test_node_attributes.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/test_drawing.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/test_repr_mimebundle.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/test_subgraph.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/test_close.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/test_edge_attributes.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/test_clear.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/test_layout.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/test_attribute_defaults.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      copying pygraphviz/tests/test_graph.py -> build/lib.macosx-14-arm64-cpython-311/pygraphviz/tests
      running egg_info
      writing pygraphviz.egg-info/PKG-INFO
      writing dependency_links to pygraphviz.egg-info/dependency_links.txt
      writing top-level names to pygraphviz.egg-info/top_level.txt
      reading manifest file 'pygraphviz.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      warning: no files found matching '*.png' under directory 'doc'
      warning: no files found matching '*.html' under directory 'doc'
      warning: no files found matching '*.txt' under directory 'doc'
      warning: no files found matching '*.css' under directory 'doc'
      warning: no previously-included files matching '*~' found anywhere in distribution
      warning: no previously-included files matching '*.pyc' found anywhere in distribution
      warning: no previously-included files matching '.svn' found anywhere in distribution
      no previously-included directories found matching 'doc/build'
      adding license file 'LICENSE'
      writing manifest file 'pygraphviz.egg-info/SOURCES.txt'
      copying pygraphviz/graphviz.i -> build/lib.macosx-14-arm64-cpython-311/pygraphviz
      copying pygraphviz/graphviz_wrap.c -> build/lib.macosx-14-arm64-cpython-311/pygraphviz
      running build_ext
      building 'pygraphviz._graphviz' extension
      creating build/temp.macosx-14-arm64-cpython-311
      creating build/temp.macosx-14-arm64-cpython-311/pygraphviz
      clang -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -I/usr/local/Cellar/graphviz/9.0.0/include/ -I/opt/homebrew/opt/openjdk/include -DSWIG_PYTHON_STRICT_BYTE_CHAR -I/Users/tom/project_folder/.venv/include -I/opt/homebrew/opt/python@3.11/Frameworks/Python.framework/Versions/3.11/include/python3.11 -c pygraphviz/graphviz_wrap.c -o build/temp.macosx-14-arm64-cpython-311/pygraphviz/graphviz_wrap.o
      clang -bundle -undefined dynamic_lookup -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX14.sdk -L/usr/local/Cellar/graphviz/9.0.0/lib/ -I/usr/local/Cellar/graphviz/9.0.0/include/ -I/opt/homebrew/opt/openjdk/include build/temp.macosx-14-arm64-cpython-311/pygraphviz/graphviz_wrap.o -lcdt -lcgraph -lgvc -o build/lib.macosx-14-arm64-cpython-311/pygraphviz/_graphviz.cpython-311-darwin.so
      ld: warning: search path '/usr/local/Cellar/graphviz/9.0.0/lib/' not found
      ld: library 'cdt' not found
      clang: error: linker command failed with exit code 1 (use -v to see invocation)
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for pygraphviz
Failed to build pygraphviz
ERROR: Could not build wheels for pygraphviz, which is required to install pyproject.toml-based projects
```

Also having it with Sonoma 14.2.1 (on M1)

Running:
```sh
brew install graphviz

pip install --use-pep517 --config-settings="--global-option=build_ext" --config-settings="--build-option=-I$(brew --prefix graphviz)/include/" --config-settings="--build-option=-L$(brew --prefix graphviz)/lib/" pygraphviz
```

Get the error

```sh
      pygraphviz/graphviz_wrap.c:3020:10: fatal error: 'graphviz/cgraph.h' file not found
      #include "graphviz/cgraph.h"
               ^~~~~~~~~~~~~~~~~~~
      1 error generated.
      error: command '/usr/bin/clang' failed with exit code 1
      [end of output]
```

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -107,7 +107,11 @@ jobs:
       run: |
         pip install --upgrade pip
         pip install -r requirements/test.txt
-        pip install .
+        pip install --use-pep517 \
+                    --config-settings="--global-option=build_ext" \
+                    --config-settings="--build-option=-I$(brew --prefix graphviz)/include/" \
+                    --config-settings="--build-option=-L$(brew --prefix graphviz)/lib/" \
+                    .
         pip list
 
     - name: Test pygraphviz
@@ -134,10 +138,10 @@ jobs:
         python-version: ${{ matrix.python-version }}
 
     - name: Download macports pkg
-      run: wget https://github.com/macports/macports-base/releases/download/v2.9.0/MacPorts-2.9.0-12-Monterey.pkg
+      run: wget https://github.com/macports/macports-base/releases/download/v2.9.3/MacPorts-2.9.3-14-Sonoma.pkg
 
     - name: Install macports
-      run: sudo installer -pkg MacPorts-2.9.0-12-Monterey.pkg -target /
+      run: sudo installer -pkg MacPorts-2.9.3-14-Sonoma.pkg -target /
 
     - name: Install graphviz
       run: sudo /opt/local/bin/port install graphviz

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -107,7 +107,7 @@ jobs:
       run: |
         pip install --upgrade pip
         pip install -r requirements/test.txt
-        pip install --use-pep517 \
+        pip install --no-cache-dir \
                     --config-settings="--global-option=build_ext" \
                     --config-settings="--build-option=-I$(brew --prefix graphviz)/include/" \
                     --config-settings="--build-option=-L$(brew --prefix graphviz)/lib/" \

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -109,8 +109,8 @@ jobs:
         pip install -r requirements/test.txt
         pip install --no-cache-dir \
                     --config-settings="--global-option=build_ext" \
-                    --config-settings="--build-option=-I$(brew --prefix graphviz)/include/" \
-                    --config-settings="--build-option=-L$(brew --prefix graphviz)/lib/" \
+                    --config-settings="--global-option=-I$(brew --prefix graphviz)/include/" \
+                    --config-settings="--global-option=-L$(brew --prefix graphviz)/lib/" \
                     .
         pip list
 

--- a/INSTALL.txt
+++ b/INSTALL.txt
@@ -76,10 +76,10 @@ Homebrew
    In this case, it may be necessary to manually specify the path to the
    graphviz include and/or library directories, e.g. ::
 
-       pip install --use-pep517 \
+       pip install --no-cache-dir \
                    --config-settings="--global-option=build_ext" \
-                   --config-settings="--build-option=-I$(brew --prefix graphviz)/include/" \
-                   --config-settings="--build-option=-L$(brew --prefix graphviz)/lib/" \
+                   --config-settings="--global-option=-I$(brew --prefix graphviz)/include/" \
+                   --config-settings="--global-option=-L$(brew --prefix graphviz)/lib/" \
                    pygraphviz
 
    See the Advanced section for details.

